p2p/event_handler.go

```diff
@@ -0,0 +1,74 @@
+package p2p
+
+import (
+	"fmt"
+
+	"github.com/pokt-network/pocket/shared/codec"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/messaging"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+func (m *p2pModule) HandleEvent(event *anypb.Any) error {
+	evt, err := codec.GetCodec().FromAny(event)
+	if err != nil {
+		return err
+	}
+
+	switch event.MessageName() {
+	case messaging.ConsensusNewHeightEventType:
+		consensusNewHeightEvent, ok := evt.(*messaging.ConsensusNewHeightEvent)
+		if !ok {
+			return fmt.Errorf("failed to cast event to ConsensusNewHeightEvent")
+		}
+
+		addrBook := m.network.GetAddrBook()
+		newAddrBook, err := m.addrBookProvider.GetStakedAddrBookAtHeight(consensusNewHeightEvent.Height)
+
+		if err != nil {
+			return err
+		}
+
+		added, removed := getAddrBookDelta(addrBook, newAddrBook)
+		for _, add := range added {
+			if err := m.network.AddPeerToAddrBook(add); err != nil {
+				return err
+			}
+		}
+		for _, rm := range removed {
+			if err := m.network.RemovePeerFromAddrBook(rm); err != nil {
+				return err
+			}
+		}
+
+	case messaging.StateMachineTransitionEventType:
+		stateMachineTransitionEvent, ok := evt.(*messaging.StateMachineTransitionEvent)
```

Comment:
NIT: Create small local helper functions for each event type. Will make it easier as we add more events

Reply:
Resolving in https://github.com/pokt-network/pocket/pull/523/files#r1109174159